runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -426,7 +426,26 @@ void CQExecutor::execute(
 void CQExecutor::execute(
     ::tt::target::metal::EnqueueWriteBufferCommand const *command) {
   ZoneScopedN("EnqueueWriteBufferCommand");
-  throw std::runtime_error("Unsupported EnqueueWriteBufferCommand");
+  assert(buffers.find(command->dst()->global_id()) != buffers.end() &&
+         "Buffer not allocated");
+  auto buffer = buffers[command->dst()->global_id()];
+  constexpr bool blocking = false;
+  switch (command->src_type()) { // currently only supporting ConstantBuffer32
+  case tt::target::metal::HostBuffer::ConstantBuffer32: {
+    auto constantBuffer32 = command->src_as_ConstantBuffer32();
+    assert(constantBuffer32->data() != nullptr &&
+           "Only constant data supported");
+    assert(command->dst()->size() ==
```

Comment:
This assert on size() is a bit questionable since it's not really comparing shape like the message says.

Reply:
Changed it to "size mismatch."